android/runtime/common/src/java/org/appcelerator/kroll/KrollRuntime.java

```diff
@@ -213,14 +213,16 @@ protected void doInit()
 
 	public void dispose()
 	{
-
-		Log.d(TAG, "Disposing runtime.", Log.DEBUG_MODE);
-
 		// Set state to released when since we have not fully disposed of it yet
 		synchronized (runtimeState) {
+			if (runtimeState == State.DISPOSED) {
+				return;
+			}
 			runtimeState = State.RELEASED;
```

Comment:
We should probably check if the runtime state is set to RELEASED as well to prevent the `internalDispose()` method from being called twice within the `dispose()` method.

```Java
if ((runtimeState == State.RELEASED) || (runtimeState == State.DISPOSED)) {
   return;
}
```

And we should add a guard to the `internalDispose()` method too...

```Java
synchronized (runtimeState) {
	if (runtimeState == State.RELAUNCHED) {
		// Abort the dispose if the application has been re-launched since we scheduled this dispose during the
		// last exit. Then set it back to the initialized state.
		runtimeState = State.INITIALIZED;
		return;
	}
	if (runtimeState == State.DISPOSED) { // <- Add this.
		return;                           // <- Add this.
	}                                     // <- Add this.
	runtimeState = State.DISPOSED;
}
```

What do you think?